src/software/geom/util.cpp

```diff
@@ -1251,6 +1116,172 @@ std::optional<Segment> mergeFullyOverlappingSegments(Segment segment1, Segment s
     }
 }
 
+std::optional<std::vector<Segment>> reduceParallelSegments(std::vector<Segment> segments)
+{
+    // If there are no segments, return nullopt
+    if (segments.size() == 0)
+    {
+        return std::nullopt;
+    }
+    else if (segments.size() == 1)
+    {
+        // If there is only 1 segments, it is unique and should be returned
+        return std::make_optional(segments);
+    }
+
+    // Check that all segments are collinear
+    for (unsigned int i = 0; i < segments.size() - 1; i++)
+    {
+        if (!collinear(segments[i], segments[i + 1]))
+        {
+            return std::nullopt;
+        }
+    }
+    std::vector<Segment> unique_segments;
+
+    unsigned int j = 0;
+    // Loop through all segments and combine segments
+    // to reduce the vector to the smallest number of independent (not overlapping)
+    // segments
+    while (segments.size() > 0)
+    {
+        std::optional<Segment> temp_segment = segments[0];
```

Comment:
Why are we casting the `Segment` to a `std::optional<Segment>` here?

Reply:
For no good reason, good catch :+1: 

Removed the casting.